### DISCUSSION TITLE
/posts/neovim-linting-and-formatting

### DISCUSSION THREAD

**Discussion by giscus** [josean-dev/sveltekit-blog#8] [General]
# /posts/neovim-linting-and-formatting

How to setup linting & formatting in Neovim with conform.nvim and nvim-lint to replace the archived null-ls plugin.

https://www.josean.com/posts/neovim-linting-and-formatting

**Comment by zeitchef** [comment]
Love the article! I see you've also integrated `none-ls` into your setup - can you comment on which is your preference? Love your contributions to here!

**Comment by evadeflow** [comment]
Thanks, this  has been a big help. I’m in the process of reworking a basic Neovim config I created almost three years ago. A *lot* has changed in that time! Your articles and videos have been really helpful in making sense of the new landscape.

**Comment by franciscohoracio** [comment]
I am very grateful to you, I spent a long time configuring my editor incorrectly.

**Comment by saitharun14** [comment]
Love the article, helped me a lot to correctly setup my linters and formatters. Keep up the good work man.

**Comment by jtuz** [comment]
Good article, it helped me with `nvim-lint` settings!

**Comment by MikeDoouglas** [comment]
Awesome article, it helped me a lot!

**Comment by studious** [comment]
Is there much value in using eslint through nvim-lint if I already have eslint-langauge-server providing linting through diagnostics?

**Comment by alexbet** [comment]
Josean, I love your articles and videos! I follow you religiously. I have a problem with your eslint_d config though. When I open a JavaScript file with multiple errors in it, I get an error message only for the first line that has an error. All other lines show no errors, as if they are fine. If I fix the error on the first line, then an error appears on the second line that has an error. Very weird! I tried the Eslint LSP instead of eslint_d, and the same thing happens. What am I doing wrong? I want to be able to see all errors inside a buffer. Just FYI, I am using a flat v9 eslint.config.mjs file for the config. The config is very simple, nothing to it really.  

Any help to solve this problem is greatly appreciated. Thanks!

**Comment by alexsasharegan** [comment]
This worked excellent for me! I ran into an issue configuring `golangci-lint`. It turns out I just had the wrong name for the lua config, not `golanci-lint`, but `golangcilint`. I guess the dash might be invalid? I'm just wondering how you figure out the name to use when even in the Mason registry is shows with the hyphen. Does anyone know?

**Comment by ninhlailaptrinh** [comment]
nice !!!!

**Comment by AntDavidLima** [comment]
Nice article, thakyou!
On the Conform.nvim section you mentioned it will look for the executable of the lint tool it is trying to use in your project dir. Can nvim-lint do it too? If not, do you have any alternative to make this happen?
